test/algolia/unit/http_requester_test.rb

```diff
@@ -0,0 +1,25 @@
+require 'algolia'
+require 'test_helper'
+
+class HttpRequesterTest
+  describe 'connection' do
+    def test_caches_http_client_connections
+      requester = Algolia::Http::HttpRequester.new(Defaults::ADAPTER, nil)
+      host1 = Algolia::Transport::StatefulHost.new('host1')
+
+      assert requester.connection(host1) == requester.connection(host1)
+      assert_equal requester.connection(host1).url_prefix.host, 'host1'
+    end
+
+    def test_caches_hosts_independent_of_each_other
+      requester = Algolia::Http::HttpRequester.new(Defaults::ADAPTER, nil)
+      host1 = Algolia::Transport::StatefulHost.new('host1')
+      host2 = Algolia::Transport::StatefulHost.new('host2')
+
+      assert requester.connection(host1) != requester.connection(host2)
```

Comment:
I think we can write it like this
```suggestion
      assert_not_equal requester.connection(host1), requester.connection(host2)
```

Reply:
Discussed in DMs, the method is not available unfortunately
```
`NoMethodError: undefined method `assert_not_equal' for #<#<Class:0x000000013d40fae0>:0x000000011da1d038>
Did you mean?  assert_equal
```